From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joshua Prince <joshua@jtprince.com>
Date: Wed, 25 Oct 2023 00:00:29 -0700
Subject: [PATCH] Amethyst cluster self-support


diff --git a/src/main/java/net/minecraft/world/level/block/AmethystClusterBlock.java b/src/main/java/net/minecraft/world/level/block/AmethystClusterBlock.java
index 01b4cf48901aaac897b8def284bf3e2c1cea3570..9974b24f35050964346d864bb202a0c7bed7482e 100644
--- a/src/main/java/net/minecraft/world/level/block/AmethystClusterBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/AmethystClusterBlock.java
@@ -63,7 +63,11 @@ public class AmethystClusterBlock extends AmethystBlock implements SimpleWaterlo
     public boolean canSurvive(BlockState state, LevelReader world, BlockPos pos) {
         Direction direction = state.getValue(FACING);
         BlockPos blockPos = pos.relative(direction.getOpposite());
-        return world.getBlockState(blockPos).isFaceSturdy(world, blockPos, direction);
+
+        // JTP Start - Amethyst self-support
+        BlockState b = world.getBlockState(blockPos);
+        return b.isFaceSturdy(world, blockPos, direction) || b.is(Blocks.AMETHYST_CLUSTER);
+        // JTP End
     }
 
     @Override
